docker/runner.go

```diff
@@ -22,52 +29,103 @@ func NewRunner() *Runner {
 }
 
 // Run the function at path
-func (n *Runner) Run(f bosonFunc.Function) error {
-	// Check for the docker binary explicitly so that we can return
-	// an extra-friendly error message.
-	_, err := exec.LookPath("docker")
+func (n *Runner) Run(ctx context.Context, f bosonFunc.Function) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 	if err != nil {
-		return errors.New("please install 'docker'")
+		return errors.Wrap(err, "failed to create docker api client")
 	}
 
-	if f.Image == "" {
-		return errors.New("Function has no associated Image. Has it been built?")
```

Comment:
I'm not seeing this check for an empty `f.Image` in the new version.  Do we no longer want to explicitly error in this case?

Reply:
good catch, we still should do the check